nmpolicy/internal/resolver/errors.go

```diff
@@ -16,10 +16,27 @@
 
 package resolver
 
-import "fmt"
+import (
+	"fmt"
 
-func pathError(format string, a ...interface{}) error {
-	return fmt.Errorf("invalid path: %v", fmt.Errorf(format, a...))
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+type PathError struct {
+	err       error
```

Comment:
Can we name this like https://github.com/nmstate/nmpolicy/blob/main/nmpolicy/internal/parser/errors.go#L25 ? 

Also we have to implement the `Unwrap` functions to support golang error unwrapping
https://github.com/nmstate/nmpolicy/blob/main/nmpolicy/internal/parser/errors.go#L29-L31